markdown/renderer.go

```diff
@@ -3,144 +3,344 @@ package markdown
 
 import (
 	"bytes"
+	"fmt"
```

Comment:
wow. great job. I haven't looked that deeply into it yet. But if the tests work, good job :D